up-l3/udiscovery/v3/service.adoc

```diff
@@ -0,0 +1,253 @@
+
+[.specitem,oft-sid="dsn~discovery-findservices-error-invalid~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.INVALID_ARGUMENT` if the URI passed does not contain a valid ue_id or a FQURI.
+--
+
+[.specitem,oft-sid="dsn~discovery-findservices-error-notfound~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.NOT_FOUND` if no service matching the URI was found.
+--
+
+[.specitem,oft-sid="dsn~discovery-findservices-error-permission~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.PERMISSION_DENIED` if the caller is not permitted to query for that service.
+--
+
+[.specitem,oft-sid="dsn~discovery-findservices-error-internal~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.INTERNAL` if an internal error occurs (database is corrupted).
+--
+
+
+
+=== FindServiceTopics()
+
+`FindServiceTopics()` is used to fetch metadata about one or more topics based on the passed `UUri`. Clients call this API to get additional information about the topic(s) such as the message format, id, permission level, etc.
+
+[.specitem,oft-sid="dsn~discovery-findservicetopics-error-invalid~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.INVALID_ARGUMENT` if the URI passed does not contain a valid ue_id or a FQURI.
+--
+
+[.specitem,oft-sid="dsn~discovery-findservicetopics-error-notfound~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.NOT_FOUND` if no topic matching the URI was found.
+--
+
+[.specitem,oft-sid="dsn~discovery-findservicetopics-error-permission~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.PERMISSION_DENIED` if the caller is not permitted to query for that topic(s).
+--
+
+[.specitem,oft-sid="dsn~discovery-findservicetopics-error-internal~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.INTERNAL` if an internal error occurs (database is corrupted).
+--
+
+=== Recursive Lookups
+
+Recursive lookups happen when the client sets the following information in the request message of the FindService() or FindServiceTopics() API. wishes to To recursively lookup up to the domain (or central) levels, the client calling the API :
+[.specitem,oft-sid="dsn~discovery-recursive-authority-wildcard~1",oft-needs="impl,test"]
+--
+*  `authority_name` *MUST* contain the wildcard `*`.
+--
+
+If `search_central` inside of the `FindServicesRequest` or `GetServiceTopicsRequest` is `true``:
+[.specitem,oft-sid="dsn~discovery-recursive-central~1",oft-needs="impl,test"]
+--
+* *MUST* search recursively up to and including the central instance.
+--
+
+UDiscovery services perform a recursive search when the local or domain instance calls `FindService()` or `FindServiceTopics()` API and then propagates the response back to the client.
+
+[.specitem,oft-sid="dsn~discovery-recursive-async~1",oft-needs="impl,test"]
+--
+* The original local search *MUST ONLY* return the response message when the recursive search has completed.
```

Comment:
FMPOV we should also shortly comment on using the TTL value appropriately for this purpose ...

Reply:
FindServicesRequest does not include a ttl, there is only the uAttributes ttl for the request.